From 62856eb8fae59ba862e7b37c3021eac367a2f286 Mon Sep 17 00:00:00 2001
From: Cui GaoSheng <cuigaosheng1@huawei.com>
Date: Wed, 27 Jul 2022 21:54:52 -0400
Subject: [PATCH 3611/4585] [Huawei] efi/libstub: arm64: support for multiple
 ',' parameters

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187251

--------------------------------

In addition, the previous code can not support multiple ','
parameter parsing , fix it.

Fixes: 3ce00bb63a80 ("[Huawei] efi/libstub: arm64: Fix KASLR and memmap= collision")
Signed-off-by: Cui GaoSheng <cuigaosheng1@huawei.com>
---
 drivers/firmware/efi/libstub/arm64-stub.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/firmware/efi/libstub/arm64-stub.c b/drivers/firmware/efi/libstub/arm64-stub.c
index 251623f97948..4638d511fb1a 100644
--- a/drivers/firmware/efi/libstub/arm64-stub.c
+++ b/drivers/firmware/efi/libstub/arm64-stub.c
@@ -55,7 +55,7 @@ void efi_parse_option_memmap(const char *str)
 	char *k, *p = (char *)str;
 
 	while (p && (idx < MAX_MEMMAP_REGIONS)) {
-		k = strchr(str, ',');
+		k = strchr(p, ',');
 		if (k)
 			*k++ = 0;
 
-- 
2.17.1

